sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -97,6 +97,13 @@ class DictVectorizer(TransformerMixin, BaseEstimator):
     array([[0., 0., 4.]])
     """
 
+    _parameter_constraints = {
+        "dtype": [type],  # TODO: TypeOptions constraint,
```

Comment:
When any dtype is valid, we'd rather delegate validation to numpy since dtype can be exressed in many different ways, as discussed in https://github.com/scikit-learn/scikit-learn/pull/23579
```suggestion
        "dtype": "n_validation"  # validation delegated to numpy
```